avocado/core/job.py

```diff
@@ -376,34 +375,16 @@ def _log_avocado_datadir():
         job_log.info('logs     ' + data_dir.get_logs_dir())
         job_log.info('')
 
-    def _log_mux_tree(self, mux):
```

Comment:
Again, I may be missing something, but I had the idea that a tree would always be present and be the source of parameters.  Looking at the diagram:

("->" means "depends on" or "knows about")

```
   +------------------------------+
   | Test                         |
   +------------------------------+
             |
             |
   +---------v---------+    +--------------------------------+
   | Parameters System |--->| Variants Generation Plugin API |
   +-------------------+    +--------------------------------+
             ^                                ^
             |                                |
   +--------------------------------------+   |
   | +--------------+ +-----------------+ |   |
   | | avocado-virt | | other providers | |   |
   | +--------------+ +-----------------+ |   |
   +--------------------------------------+   |
                                              |
                 +----------------------------+-----+
                 |                                  |
                 |                                  |
                 |                                  |
       +--------------------+           +-------------------------+
       | Multiplexer Plugin |           | Other variant plugin(s) |
       +--------------------+           +-------------------------+
             |
             |
       +-----v---------------------------+
       | +------------+ +--------------+ |
       | | --mux-yaml | | --mux-inject | |
       | +------------+ +--------------+ |
       +---------------------------------+
```

Isn't the parameter system backed by a Tree? The only other option I can see, which looks incomplete, is if `default_params` would be (but is not) a replacement for it.